### PR TITLE
GEODE-9969: Fix unescaping the region name with underscore

### PR DESCRIPTION
**Pull request by mkevo**
### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

File: geode-core/src/test/java/org/apache/geode/internal/cache/PartitionedRegionHelperJUnitTest.java [hunk 1]
```diff
@@ -41,4 +41,22 @@ public void testEscapeUnescape() {
     }
   }
 
+  @Test
+  public void testEscapeUnescapeWhenRegionHasUnderscoreInTheName() {
+    {
```
Comment (DonalEvans) [inline geode-core/src/test/java/org/apache/geode/internal/cache/PartitionedRegionHelperJUnitTest.java:46]: Can these unnecessary braces be removed please?

**Review by DonalEvans — CHANGES_REQUESTED** [review]
A few small clean-up requests. Also, if it's not too difficult, would it be possible to get a DUnit test showing the behaviour described in the ticket? It would be good to have confirmation in a test that the user-visible effect of this change is correct.

If it's too difficult to reproduce the issue in a DUnit test then that's okay though.

**Comment by mkevo** [comment]
> A few small clean-up requests. Also, if it's not too difficult, would it be possible to get a DUnit test showing the behaviour described in the ticket? It would be good to have confirmation in a test that the user-visible effect of this change is correct.
> 
> If it's too difficult to reproduce the issue in a DUnit test then that's okay though.

Hi @DonalEvans,
Thank you for the fast response!
I tried to create DUnit test, but without success.
The main problem with test is how to rename the working dir of the killed server to some other name, so it can be used as a working dir for the new server. 
If you have some idea to do it, please feel free to share it. If not, we also have this test that verifies the solution.

**Comment by DonalEvans** [comment]
@mkevo I might be wrong, but I think it should be possible to get a new server to use the working directory of a shutdown server using the `--dir` option on the `start server` gfsh command. By default, the directory is named after the server, but you should be able to do something like:
```
start server --name=server1 --server-port=40401

***kill the server***

start server --name=server2 --dir=server1 --server-port=40402
```

**Comment by kirklund** [comment]
@DonalEvans @mkevo `--dir` won't work in a distributed test (dunit). It will however work in an acceptance test. Take a look at the tests under:
* [geode-assembly/src/acceptanceTest/java/org/apache/geode/serialization/filter/](https://github.com/apache/geode/tree/develop/geode-assembly/src/acceptanceTest/java/org/apache/geode/serialization/filter)
* [geode-assembly/src/acceptanceTest/java/org/apache/geode/logging/](https://github.com/apache/geode/tree/develop/geode-assembly/src/acceptanceTest/java/org/apache/geode/logging)

And (these are older tests so probably not as good):
* [geode-assembly/src/acceptanceTest/java/org/apache/geode/launchers/](https://github.com/apache/geode/tree/develop/geode-assembly/src/acceptanceTest/java/org/apache/geode/launchers)

You can get a reference to the `Process` (to kill it) from `GfshExecution.getProcess()`. `GfshExecution` for that server is returned from `gfshRule.execute(startServerCommand);`

> @mkevo I might be wrong, but I think it should be possible to get a new server to use the working directory of a shutdown server using the `--dir` option on the `start server` gfsh command. By default, the directory is named after the server, but you should be able to do something like:
> 
> ```
> start server --name=server1 --server-port=40401
> 
> ***kill the server***
> 
> start server --name=server2 --dir=server1 --server-port=40402
> ```

**Review by dschneider-pivotal — APPROVED** [review]

**Review by DonalEvans — APPROVED** [review]

**Pull request merged**